src/MediatR/Mediator.cs

```diff
@@ -37,48 +39,76 @@ public TResponse Send<TResponse>(IRequest<TResponse> request)
         {
             var defaultHandler = GetHandler(request);
 
-            var result = defaultHandler.Handle(request);
+            var pipeline = GetPipeline(request, () => Task.FromResult(defaultHandler.Handle(request)));
 
-            return result;
+            return pipeline.Result;

```

Comment:
Yep. Async over sync is fiiine. Sync over async, not so much :cry: Hence [my comment](https://github.com/jbogard/MediatR/issues/102#issuecomment-266003071) :wink: